src/main/java/org/peergos/util/TraceLogger.java

```diff
@@ -253,25 +260,39 @@ private TraceLogger() {
         nodeIdMap.put("12D3KooWCfWmdJYdAUwm1pTxFKsRDRzGmsbeUMpriNNocMDVMmum", 9);
         currentNodeId = -1;
 
+        createAndSetupOutput();
+
+    }
+
+    private void createAndSetupOutput() {
         try {
-            outFile = new FileOutputStream(Client.DEFAULT_IPFS_DIR_PATH.toAbsolutePath().toString() + "/trace.log",
+            // Prepend each log file with the creation timestamp
+            long logFileCreationTimestampMillis = System.currentTimeMillis();
+
+            outFile = new FileOutputStream(
+                    Client.DEFAULT_IPFS_DIR_PATH.toAbsolutePath().toString() + "/" + logFileCreationTimestampMillis
+                            + ".trace.log",
                     /* append= */ true);
+
+            // An output stream with 100 MB buffer.
+            output = new BufferedOutputStream(outFile, 100 * 1024 * 1024);
```

Comment:
@sonudoo I think this should match the `MAX_LOG_FILE_SIZE_MB` but just wanted to double check

Reply:
Yes that makes sense.